src/commonMain/kotlin/character/classes/priest/pet/abilities/Melee.kt

```diff
@@ -0,0 +1,98 @@
+package character.classes.priest.pet.abilities
+
+import character.Ability
+import character.auto.AutoAttackBase
+import character.classes.priest.Priest
+import character.classes.priest.pet.Shadowfiend
+import character.Proc
+import character.Resource
+import character.Stats
+import data.Constants
+import data.model.Item
+import data.model.ItemSet
+import data.model.Socket
+import data.model.SocketBonus
+import mechanics.Spell
+import mechanics.Melee
+import sim.SimParticipant
+import sim.*
+
+class Melee : AutoAttackBase() {
+    companion object {
+        const val name = "Melee"
+        const val manaRestoreName = "Shadowfiend"
+    }
+    override val id: Int = 1
+    override val name: String = Companion.name
+    override fun gcdMs(sp: SimParticipant): Int = 1500
+
+    override fun castTimeMs(sp: SimParticipant): Int = 0
+    override fun resourceCost(sp: SimParticipant): Double {
+        return sp.character.klass.baseMana * 0.01
+    }
+
+    val school = Constants.DamageType.SHADOW
+    override fun item(sp: SimParticipant): Item {
+        return object : Item() {
+            override var isAutoGenerated: Boolean = false
+            override var id: Int = -1
+            override var name: String = "Melee (Pet)"
+            override var itemLevel: Int = 1
+            override var quality: Int = 1
+            override var icon: String = ""
+            override var itemSet: ItemSet? = null
+            override var inventorySlot: Int = 21
+            override var itemClass: Constants.ItemClass? = null
+            override var itemSubclass: Constants.ItemSubclass? = null
+            override var minDmg: Double = 99.0 + 0.12 * (sp.stats.attackPower.toDouble() - 286.0)
```

Comment:
Possibly worth constraining these to a minimum of zero, as there is some potential it could be negative